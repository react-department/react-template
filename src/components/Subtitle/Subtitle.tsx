import { testIdSubtitle } from '../../constants/TestId';

import type { ReactElement } from 'react';
import type ISubtitle from './interfaces/ISubtitle';

import styles from './sass/Subtitle.module.scss';

/**
 * H2 subtitle component
 */
export default function Subtitle({ text }: ISubtitle): ReactElement {
  return (
    <h2 className={styles.appSubtitle} data-testid={testIdSubtitle}>
      {text}
    </h2>
  );
}
