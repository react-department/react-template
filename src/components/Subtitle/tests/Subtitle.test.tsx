import { screen } from '@testing-library/react';

import { render } from '../../../../__tests__/test-utils';
import Subtitle from '../Subtitle';

describe('<Subtitle />', () => {
  test('should render component', () => {
    const { asFragment } = render(<Subtitle text="Subtitle" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render component with title', () => {
    render(<Subtitle text="Subtitle" />);
    expect(screen.getByText('Subtitle')).toBeInTheDocument();
  });
});
