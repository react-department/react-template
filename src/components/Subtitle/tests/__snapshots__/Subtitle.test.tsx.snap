// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subtitle /> should render component 1`] = `
<DocumentFragment>
  <h2
    class="appSubtitle"
    data-testid="subtitle"
  >
    Subtitle
  </h2>
</DocumentFragment>
`;
